api/src/auth/__tests__/auth.service.spec.ts

```diff
@@ -0,0 +1,301 @@
+import type { User } from '@/users/schemas/user.schema';
+import { UsersService } from '@/users/users.service';
+
+import { AuthService } from '../auth.service';
+import { createAccountDtoStubFactory } from './stubs/create-account.dto.stub';
+import { currentUserStubFactory } from './stubs/current-user.stub';
+import { verifyAccountStubFactory } from './stubs/verify-account.dto.stub';
+
+const validDate = Date.now() + 360000;
+const expiredDate = Date.now() - 10000;
+
+describe('AuthService', () => {
+  let authService: AuthService;
+  let usersService: MockedInstance<UsersService>;
+  let cryptoService: MockedInstance<CryptoService>;
+  let jwtService: MockedInstance<JwtService>;
+
+  // called before every test to intialize the AuthService
+  beforeEach(async () => {
+    const moduleRef = await Test.createTestingModule({
+      providers: [
+        AuthService,
+        {
+          provide: ConfigService,
+          useValue: {
+            getOrThrow: (propertyPath: string) => propertyPath
+          }
+        },
+        {
+          provide: CryptoService,
+          useValue: createMock(CryptoService)
+        },
+        {
+          provide: I18nService,
+          useValue: createMock(I18nService)
+        },
+        { provide: JwtService, useValue: createMock(JwtService) },
+        {
+          provide: MailService,
+          useValue: createMock(MailService)
+        },
+        {
+          provide: UsersService,
+          useValue: createMock(UsersService)
+        }
+      ]
+    }).compile();
+    // get all the services we need from the moduleRef
+    authService = moduleRef.get(AuthService);
+    usersService = moduleRef.get(UsersService);
+    cryptoService = moduleRef.get(CryptoService);
+    jwtService = moduleRef.get(JwtService);
+  });
+
+  it('should be defined', () => {
+    expect(authService).toBeDefined();
+  });
+
+  let createAccountDto: CreateAccountDto;
+  let createUserDto: CreateUserDto;
+  let verifyAccountDto: VerifyAccountDto;
+  let currentUser: CurrentUser;
+
+  beforeEach(() => {
+    createAccountDto = createAccountDtoStubFactory();
+    createUserDto = createUserDtoStubFactory();
+    verifyAccountDto = verifyAccountStubFactory();
+    currentUser = currentUserStubFactory();
+    jwtService.signAsync.mockResolvedValue('accessToken');
+  });
+
+  describe('createAccount', () => {
+    it('calls the usersService.createUser and returns the created account', async () => {
+      const mockUser = {
+        ...createAccountDto,
+        isVerified: false,
+        role: 'standard'
+      };
+      usersService.createUser.mockResolvedValue(mockUser);
+      const result = await authService.createAccount(createAccountDto);
+      expect(result).toMatchObject(mockUser);
+    });
+  });
+
+  describe('login', () => {
+    beforeEach(() => {
+      // can use spyOn insead: spyOn(usersService, 'findByEmail').mockResolvedValue(createUserDto);
+      usersService.findByEmail.mockResolvedValue(createUserDto);
```

Comment:
Remember to clear your mocks after you are done, see: https://bun.sh/docs/test/mocks